adobe/commerce-cif-magento
# CIF-452 Healthcheck implementation for Magento

## Description
Still in progress writing tests
DON'T MERGE

As a developer/devops engineer I want the health check API to be implemented for the Magento connector.

The implementation must respect the design done by CIF-451.

Deliverables

Magento code updated and the healthcheck API available.
Updated documentation.
Acceptance criteria

The health check connects to the commerce backend when deciding if the system is healthy or not. Check the parent epic for more information on what to verify from the commerce backend.
Do not use web action transformer for this.
We have comprehensive ITs for this. Use a mock server which we can configure how to respond or it will respond at all.
Verification steps

Setup:

**Deploy the Magento connector which contains the healthcheck implementation.
Steps:

Make sure system is healthy. Call the healthcheck and make sure it reports the system is healthy.
Break the commerce backend. Call the healthcheck and make sure it reports that the system is NOT healthy and provides some information about what part of it is not healthy. 


## Related Issue






## Motivation and Context



## How Has This Been Tested?





## Screenshots (if appropriate):

## Types of changes




## Checklist:




- [x] My code passes the code style as defined by ESLint.
